sdk/purview/purview-workflow-rest/README.md

```diff
@@ -0,0 +1,140 @@
+# Azure Purview Workflow Rest-Level client library for JavaScript
+
+Workflows are automated, repeatable business processes that users can create within Microsoft Purview to validate and orchestrate CUD (create, update, delete) operations on their data entities. Enabling these processes allow organizations to track changes, enforce policy compliance, and ensure quality data across their data landscape.
+
+Use the client library for Purview Workflow to:
+
+- Manage workflows
+- Submit user requests and monitor workflow runs
+- View and respond to workflow tasks
+
+**For more details about how to use workflow, please refer to the [service documentation][product_documentation]**
+
+## Getting started
+
+### Currently supported environments
+
+- Node.js version 14.x.x or higher
+
+### Prerequisites
+
+- You must have an [Azure subscription][azure_subscription] and a [Purview resource][purview_resource] to use this package.
+
+### Create and authenticate a `PurviewWorkflowClient`
+
+Since the Workflow service uses an Azure Active Directory (AAD) bearer token for authentication and identification, an email address should be encoded into the token to allow for notification when using Workflow. It is recommended that the [Azure Identity][azure_identity] library be used  with a the [UsernamePasswordCredential][username_password_credential]. Before using the [Azure Identity][azure_identity] library with Workflow, [an application][app_registration] should be registered and used for the clientId passed to the [UsernamePasswordCredential][username_password_credential].
+Set the values of the client ID, tenant ID, username and password as environment variables:
+AZURE_CLIENT_ID, AZURE_TENANT_ID, USERNAME, PASSWORD
+
+```typescript
+import PurviewWorkflow from "@azure-rest/purview-workflow";
+import { UsernamePasswordCredential } from "@azure/identity";
```

Comment:
Does purview workloads support other credential types ?  Not sure if it's a good idea to recommend customer to use UsernamePasswordCredential in README.md @xirzec

Reply:
Workflow need to get username or email address from AAD token, other credential doesn't encode such information into token, so if user use other credential to call Workflow APIs, it will leads to some APIs call failed.

Reply:
@joheredi @xirzec Can you help confirm if it's expected behavior for other credential type in @azure/identity ?

Reply:
It seems to be consistent across languages. @CordeliaGrey would you mind sharing service docs that talk about the need of user name and password for me to get more context please?

Reply:
Here is the service doc link: https://learn.microsoft.com/en-us/azure/purview/concept-workflow
But the doc just shows how to use Workflow in portal, it didn't talk about  the requirement of username and password, the context is Workflow service need to get username(typically it is an email address) to send email notification to someone, so for SDK client, it should authenticate as users, according to the [credential classes](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity#authenticate-users), we think usernamepassword credential is more appropriate in such scenario.